doc/shared/suncontrol/SUNControl_Description.rst

```diff
@@ -0,0 +1,480 @@
+   of previous *dsm* or *h* values. The return value is an integer flag denoting
+   success/failure of the routine (see
+   :numref:`SUNControl.Description.errorCodes`).
+
+   :param C:  the :c:type:`SUNControl` object..
+   :return: error code indicating success failure
+            (see :numref:`SUNControl.Description.errorCodes`).
+
+   Usage:
+
+   .. code-block:: c
+
+      retval = SUNControlReset(C);
+
+.. c:function:: int SUNControlSetDefaults(SUNControl C)
+
+   Sets the controller parameters to their default values.
+
+   :param C:  the :c:type:`SUNControl` object..
+   :return: error code indicating success failure
+            (see :numref:`SUNControl.Description.errorCodes`).
+
+   Usage:
+
+   .. code-block:: c
+
+      retval = SUNControlSetDefaults(C);
+
+.. c:function:: int SUNControlWrite(SUNControl C, FILE* fptr)
+
+   Writes all controller parameters to the indicated file pointer.
+
+   :param C:  the :c:type:`SUNControl` object..
+   :param fptr:  the output stream to write the parameters.
+   :return: error code indicating success failure
+            (see :numref:`SUNControl.Description.errorCodes`).
+
+   Usage:
+
+   .. code-block:: c
+
+      retval = SUNControlWrite(C, stdout);
+
+.. c:function:: int SUNControlSetMethodOrder(SUNControl C, int q)
+
+   Called by the time integrator to inform the controller of the asymptotic
+   order of accuracy for the method.
+
+   :param C:  the :c:type:`SUNControl` object..
+   :param q:  the asymptotic order of accuracy for the time integration method.
+   :return: error code indicating success failure
+            (see :numref:`SUNControl.Description.errorCodes`).
+
+   Usage:
+
+   .. code-block:: c
+
+      retval = SUNControlSetMethodOrder(C, 3);
+
+.. c:function:: int SUNControlSetEmbeddingOrder(SUNControl C, int p)
+
+   Called by the time integrator to inform the controller of the asymptotic
+   order of accuracy for the method embedding.
+
+   :param C:  the :c:type:`SUNControl` object..
+   :param p:  the asymptotic order of accuracy for the time integration method
+              embedding.
+   :return: error code indicating success failure
+            (see :numref:`SUNControl.Description.errorCodes`).
+
+   Usage:
+
+   .. code-block:: c
+
+      retval = SUNControlSetEmbeddingOrder(C, 2);
+
+.. c:function:: int SUNControlSetErrorBias(SUNControl C, realtype bias)
+
+   Sets an error bias factor for scaling the local error factors. This is
+   typically used to slightly exaggerate the temporal error during the
+   estimation process, leading to a more conservative estimated step size.
+
+   :param C:  the :c:type:`SUNControl` object..
+   :param bias:  the error bias factor.
+   :return: error code indicating success failure
+            (see :numref:`SUNControl.Description.errorCodes`).
+
+   Usage:
+
+   .. code-block:: c
+
+      retval = SUNControlSetErrorBias(C, 1.2);
+
+.. c:function:: int SUNControlUpdate(SUNControl C, realtype h, realtype dsm)
```

Comment:
Great question, and I bet that it would be needed for something like CVODE or IDA.  That said, I would prefer those to use a different update function so as not to over-complicate things (e.g., a variable-order MRI controller would then also need P and p added to the list).  I recommend that we expand this API when/if we create H-p controllers, and since all implementations use `NewEmpty` then those updates wouldn't require major version number changes.